loadbalancer/zk/publisher_test.go

```diff
@@ -43,13 +44,16 @@ func TestBadFactory(t *testing.T) {
 	}
 	defer p.Stop()
 
+	// instance1 came online
+	client.AddService(path+"/instance1", "zookeeper_node_data")
+	time.Sleep(1 * time.Millisecond)
```

Comment:
Even 2ms will likely be too short on Windows once we are on Go 1.6. From the 1.6 release notes:

>On Windows, Go programs in Go 1.5 and earlier forced the global Windows timer resolution to 1ms at startup by calling timeBeginPeriod(1). Go no longer needs this for good scheduler performance, and changing the global timer resolution caused problems on some systems, so the call has been removed.

The default timer resolution on Windows is 15.6ms. Here is an informative post about the details: https://randomascii.wordpress.com/2013/04/02/sleep-variation-investigated/.

The Go issue discussion is lengthy and informative as well: https://github.com/golang/go/issues/8687

This upcoming change also means that doing something like:

```go
start := time.Now()
// do work
elapsed := time.Since(start)
```

Will measure durations in 1/64s quanta on Windows. Which means that if *work* only takes a few milliseconds, `elapsed` will usually be zero.

Avoiding sleep like functions for tests is ideal, but when that is not practical I recommend sleeping at least 40ms so the test is more likely to work as intended on Windows.